File: frontend/placement-frontend/src/services/companyService.js
import api from './api';

/**
 * Service module for Company API operations.
 * All methods return Axios promise responses.
 */

const BASE = '/companies';

const companyService = {
  /**
   * Fetch paginated companies, optionally filtered by name search.
   * @param {Object} params - { search, page, size, sort }
   */
  getAll: (params = {}) => {
    const { search, page = 0, size = 10, sort = 'name,asc' } = params;
    return api.get(BASE, { params: { search, page, size, sort } });
  },

  /**
   * Fetch a single company by ID.
   * @param {number} id
   */
  getById: (id) => api.get(`${BASE}/${id}`),

  /**
   * Create a new company.
   * @param {Object} data - Company form data
   */
  create: (data) => api.post(BASE, data),

  /**
   * Update an existing company.
   * @param {number} id
   * @param {Object} data - Updated company form data
   */
  update: (id, data) => api.put(`${BASE}/${id}`, data),

  /**
   * Delete a company by ID.
   * @param {number} id
   */
  delete: (id) => api.delete(`${BASE}/${id}`),
};

export default companyService;

File: backend/placement-management/src/main/java/com/placement/management/service/CompanyService.java
package com.placement.management.service;

import com.placement.management.dto.CompanyRequest;
import com.placement.management.dto.CompanyResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Service interface for {@link com.placement.management.entity.Company} business logic.
 *
 * @author Team — feature/company-drive
 */
public interface CompanyService {

    /**
     * Creates a new company.
     *
     * @param request the company details
     * @return the persisted company as a response DTO
     */
    CompanyResponse create(CompanyRequest request);

    /**
     * Updates an existing company by ID.
     *
     * @param id      the company's primary key
     * @param request the updated company details (null fields are ignored)
     * @return the updated company as a response DTO
     */
    CompanyResponse update(Long id, CompanyRequest request);

    /**
     * Deletes a company by ID.
     *
     * @param id the company's primary key
     */
    void delete(Long id);

    /**
     * Fetches a single company by ID.
     *
     * @param id the company's primary key
     * @return the company response DTO
     */
    CompanyResponse findById(Long id);

    /**
     * Returns a paginated list of all companies.
     *
     * @param pageable pagination and sorting parameters
     * @return page of company response DTOs
     */
    Page<CompanyResponse> findAll(Pageable pageable);

    /**
     * Searches companies by name (case-insensitive, partial match).
     *
     * @param name     the search term
     * @param pageable pagination and sorting parameters
     * @return page of matching company response DTOs
     */
    Page<CompanyResponse> search(String name, Pageable pageable);
}
